src/Microsoft.TemplateEngine.Edge/TemplateConstraintManager.cs

```diff
@@ -24,7 +24,7 @@ public TemplateConstraintManager(IEngineEnvironmentSettings engineEnvironmentSet
             _logger.LogDebug($"Found {constraintFactories.Count()} constraints factories, initializing.");
             foreach (var constraintFactory in constraintFactories)
             {
-                _templateConstrains[constraintFactory.Type] = Task.Run(() => constraintFactory.CreateTemplateConstraintAsync(engineEnvironmentSettings, _cancellationTokenSource.Token));
+                _templateConstrains[constraintFactory.Type] = constraintFactory.CreateTemplateConstraintAsync(engineEnvironmentSettings, _cancellationTokenSource.Token);
```

Comment:
I've been staring at this for a few minutes. My thoughts are... it is still running this asynchronously, since there is no `await`. The difference is that without `Task.Run`, it isn't part of a thread pool; it runs each asynchronously on the same thread.

I don't know the exact issue here, but I did notice one thing. The original `Task.Run` call forgot to pass the cancellation toaken. Meaning, it could have been:
> Task.Run(() => constraintFactory.CreateTemplateConstraintAsync(engineEnvironmentSettings, _cancellationTokenSource.Token), _cancellationTokenSource.Token);